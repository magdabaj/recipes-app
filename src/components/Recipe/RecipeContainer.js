import React from 'react';
import styled from 'styled-components';
import colorStyles from '../../utils/colorStyles';

const RecipeContainer = styled.div`
  display: flex;
  flex-direction: row;
  margin-bottom: 3vw;
  color: ${colorStyles.darkRed};

  @media (max-width: 768px) {
    justify-content: center;
  }
`;

export default RecipeContainer;
