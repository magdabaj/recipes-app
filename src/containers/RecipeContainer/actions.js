/*
 *
 * RecipeContainer actions
 *
 */

import {
    GET_RATINGS_ERROR,
    GET_RATINGS,
    GET_RATINGS_SUCCESS,
    GET_RECIPE,
    GET_RECIPE_ERROR,
    LOAD_TAGS,
    LOAD_TAGS_SUCCESS,
    LOAD_TAGS_ERROR,
    GET_RECIPE_SUCCESS,
    SEND_RATING,
    SEND_RATING_ERROR,
    SEND_RATING_SUCCESS,
    GET_COMMENTS,
    GET_COMMENTS_SUCCESS,
    GET_COMMENTS_ERROR,
    ADD_COMMENT,
    ADD_COMMENT_SUCCESS,
    ADD_COMMENT_ERROR,
    EDIT_COMMENT,
    EDIT_COMMENT_SUCCESS,
    EDIT_COMMENT_ERROR,
    REMOVE_COMMENT,
    REMOVE_COMMENT_SUCCESS,
    REMOVE_COMMENT_ERROR, CLEAR_STATUS, CLEAR_DELETE_COMMENTS_STATUS,
} from './constants';

export const getRecipeRatings = recipeId => ({
    type: GET_RATINGS,
    recipeId,
});

export const getRecipeRatingsSuccess = (message, recipeId, ratingsMean) => ({
    type: GET_RATINGS_SUCCESS,
    message,
    recipeId,
    ratingsMean,
});

export const getRecipeRatingsError = error => ({
    type: GET_RATINGS_ERROR,
    error,
});

export const getRecipe = recipeId => ({
    type: GET_RECIPE,
    recipeId,
});

export const getRecipeSuccess = recipe => ({
    type: GET_RECIPE_SUCCESS,
    recipe,
});

export const getRecipeError = error => ({
    type: GET_RECIPE_ERROR,
    error,
});


export const sendRating = (rate, userId, recipeId) => ({
    type: SEND_RATING,
    rate,
    userId,
    recipeId,
});

export const sendRatingSuccess = (message, recipeId, ratingsMean) => ({
    type: SEND_RATING_SUCCESS,
    message,
    recipeId,
    ratingsMean,
});

export const sendRatingError = error => ({
    type: SEND_RATING_ERROR,
    error,
});

export const getComments = recipeId => ({
    type: GET_COMMENTS,
    recipeId,
});

export const getCommentsSuccess = response => ({
    type: GET_COMMENTS_SUCCESS,
    response,
});

export const getCommentsError = error => ({
    type: GET_COMMENTS_ERROR,
    error,
});

export const addComment = (recipeId, content, userId) => ({
    type: ADD_COMMENT,
    recipeId,
    userId,
    content,
});

export const addCommentSuccess = message => ({
    type: ADD_COMMENT_SUCCESS,
    message,
});

export const addCommentError = error => ({
    type: ADD_COMMENT_ERROR,
    error,
});

export const editComment = (commentId, content, userId) => ({
    type: EDIT_COMMENT,
    commentId,
    userId,
    content,
});

export const editCommentSuccess = message => ({
    type: EDIT_COMMENT_SUCCESS,
    message,
});

export const editCommentError = error => ({
    type: EDIT_COMMENT_ERROR,
    error,
});

export const removeComment = commentId => ({
    type: REMOVE_COMMENT,
    commentId,
});

export const removeCommentSuccess = message => ({
    type: REMOVE_COMMENT_SUCCESS,
    message,
});

export const removeCommentError = error => ({
    type: REMOVE_COMMENT_ERROR,
    error,
});

export const clearStatus = () => ({
    type: CLEAR_STATUS
})

export const clearDeleteCommentsStatus = () => ({
    type: CLEAR_DELETE_COMMENTS_STATUS
})
