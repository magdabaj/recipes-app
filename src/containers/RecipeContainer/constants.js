/*
 *
 * RecipeContainer constants
 *
 */

export const GET_RATINGS = 'app/RecipeContainer/GET_RATINGS';
export const GET_RATINGS_SUCCESS = 'app/RecipeContainer/GET_RATINGS_SUCCESS';
export const GET_RATINGS_ERROR = 'app/RecipeContainer/GET_RATINGS_ERROR';

export const GET_RECIPE = 'app/RecipeContainer/GET_RECIPE';
export const GET_RECIPE_SUCCESS = 'app/RecipeContainer/GET_RECIPE_SUCCESS';
export const GET_RECIPE_ERROR = 'app/RecipeContainer/GET_RECIPE_ERROR';

export const LOAD_TAGS = 'app/RecipeContainer/LOAD_TAGS';
export const LOAD_TAGS_SUCCESS = 'app/RecipeContainer/LOAD_TAGS_SUCCESS';
export const LOAD_TAGS_ERROR = 'app/RecipeContainer/LOAD_TAGS_ERROR';

export const SEND_RATING = 'app/RecipeContainer/SEND_RATING';
export const SEND_RATING_SUCCESS = 'app/RecipeContainer/SEND_RATING_SUCCESS';
export const SEND_RATING_ERROR = 'app/RecipeContainer/SEND_RATING_ERROR';

export const GET_COMMENTS = 'app/RecipeContainer/GET_COMMENTS';
export const GET_COMMENTS_SUCCESS = 'app/RecipeContainer/GET_COMMENTS_SUCCESS';
export const GET_COMMENTS_ERROR = 'app/RecipeContainer/GET_COMMENTS_ERROR';

export const ADD_COMMENT = 'app/RecipeContainer/ADD_COMMENT';
export const ADD_COMMENT_SUCCESS = 'app/RecipeContainer/ADD_COMMENT_SUCCESS';
export const ADD_COMMENT_ERROR = 'app/RecipeContainer/ADD_COMMENT_ERROR';

export const EDIT_COMMENT = 'app/RecipeContainer/EDIT_COMMENT';
export const EDIT_COMMENT_SUCCESS = 'app/RecipeContainer/EDIT_COMMENT_SUCCESS';
export const EDIT_COMMENT_ERROR = 'app/RecipeContainer/EDIT_COMMENT_ERROR';

export const REMOVE_COMMENT = 'app/RecipeContainer/REMOVE_COMMENT';
export const REMOVE_COMMENT_SUCCESS = 'app/RecipeContainer/REMOVE_COMMENT_SUCCESS';
export const REMOVE_COMMENT_ERROR = 'app/RecipeContainer/REMOVE_COMMENT_ERROR';
