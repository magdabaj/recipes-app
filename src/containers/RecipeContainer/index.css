.comments-container {
    /*display: flex;*/
    /*flex-direction: row;*/
    margin-bottom: 3vw;
    margin: 10px;
    /*color: #5D001E;*/
}

.comments-list {
    list-style: none;
    /*margin: 50px 0 0;*/
    padding: 0;
}

.comment-body {
    display: list-item;
    text-align: -webkit-match-parent;
}

.comment-article {
    /*position: relative;*/
    margin: 25px 0;
    /*border-bottom: 1px solid #9A1750;*/
    box-shadow: inset 0 0 0 1px rgba(0, 0, 0, 0.1);
    padding: 20px;
}

.comment-edit {
    display: flex;
    font-size: 12px;
    font-weight: 600;
}

.comment-edit--button {
    margin-right: 12px;
    transition: 0.2s ease-in-out;
    cursor: pointer;
}

.comment-edit--button:hover {
    color: #9a1750;
}
.comment-title {
    margin: 0;
    padding: 0;
    display: flex;
    justify-content: space-between;
    flex-wrap: wrap;
    -webkit-box-pack: justify;
}

.comment-author {
    width: 50%;
    font-weight: 700;
    font-size: 1.5vw;
    text-align: left;
}

.comment-data {
    width: 50%;
    opacity: 0.3;
    font-size: 1vw;
}

.comment-content {
    font-size: 1.5vw;
    color: #303030;
    margin: 10px 0;
    text-align: left;
}

/*comments form component*/
.comment-form-container {
    margin: 25px 0;
}
.comment-text-area {
    width: 100%;
    border-radius: 0;
    border: 1px solid #c7c7c7;
    outline: none;
    padding: 15px;
    color: #303030;
    margin-bottom: 20px;
}

.comment-label {
    display: block;
    font-weight: 900;
    font-size: 2.5vw;
    text-align: left;
    margin: 50px 0;
}

.comment-button {
    background-color: #9a1750;
    color: white;
    border: 1px solid #9a1750;
    text-transform: uppercase;
    font-size: 15px;
    font-weight: 500;
    padding: 15px 25px;
    cursor: pointer;
    display: flex;
    align-items: flex-start;
}

.comment-button:hover {
    background-color: white;
    color: #9a1750;
}

.pagination-list {
    display: flex;
    list-style-type: none;
    justify-content: center;
    -webkit-box-pack: center;
    margin: 0 auto;
    padding: 0;
}

.pagination-list li {
    margin: 0 6px;
}

.pagination-link {
    cursor: pointer;
    font-size: 16px;
    font-weight: 700;
    color: #303030;
    width: 36px;
    height: 36px;
    border: 1px solid #505050;
    border-radius: 50%;
    display: flex;
    justify-content: center;
    align-items: center;
    text-align: center;
}

.pagination-link:hover {
    background-color: #9a1750;
    color: #ffffff;
    border: 1px solid #9a1750;
}


@media only screen and (max-width: 768px) {
    .comment-label {
        margin: 20px 0;
        font-size: 4.5vw;
    }
    .comment-button {
        width: 100%;
        justify-content: center;
    }

    .comment-article{
        margin: 0;
        padding: 10px;
    }

    .comment-author {
        font-size: 3vw;
    }

    .comment-data {
        font-size: 2.5vw;
    }

    .comment-content {
        font-size: 2.5vw;
    }
    .container {
        align-items: center;
    }
    .wrapper {
        flex-direction: column;
        align-items: center;
        justify-content: center;
        width: 100%;
    }

    .recipes-container {
        width: 100%;
    }
    .tags-container {
        width: 100%;
    }
}

.container {
    display: flex;
    flex-direction: row;
    /*align-content: center;    \*/
    justify-content: center;
    padding: 20px;
}

@media only screen and (min-width: 769px) {
    .recipes-container {
        width: calc(100% - 300px);
    }
    .tags-container {
        width: 300px;
    }
}

.wrapper {
    display: flex;
    max-width: 1170px;
}

.recipes-container {
    padding: 2vw;
    /*margin-left: 8vw;*/
}

.tags-container {
    padding: 2vw;
    /*margin: 4px;*/
    text-align: left;
    list-style-type: none;
    margin-block-start: 0;
}

.tag-type {
    cursor: pointer;
    font-weight: bold;
    transition: 0.3s;
}

.tag-type:hover {
    color: #9a1750;
}

.empty-div {
    width: 5%;
}
