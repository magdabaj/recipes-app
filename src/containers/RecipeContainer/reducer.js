/*
 *
 * RecipeContainer reducer
 *
 */
import produce from 'immer';
import {
    GET_RECIPE,
    GET_RECIPE_ERROR,
    GET_RECIPE_SUCCESS,
    GET_RATINGS,
    GET_RATINGS_ERROR,
    GET_RATINGS_SUCCESS,
    LOAD_TAGS,
    LOAD_TAGS_ERROR,
    LOAD_TAGS_SUCCESS,
    SEND_RATING,
    SEND_RATING_ERROR,
    SEND_RATING_SUCCESS,
    GET_COMMENTS,
    GET_COMMENTS_SUCCESS,
    GET_COMMENTS_ERROR,
    ADD_COMMENT,
    ADD_COMMENT_SUCCESS,
    ADD_COMMENT_ERROR,
    EDIT_COMMENT,
    REMOVE_COMMENT,
    REMOVE_COMMENT_SUCCESS,
    REMOVE_COMMENT_ERROR, CLEAR_STATUS, CLEAR_DELETE_COMMENTS_STATUS, CLEAR_SEND_RATING_STATUS,
} from './constants';
import fetchStates from '../../utils/fetchStates';

export const initialState = {
    ratingsMean: null,
    recipe: [],
    recipeId: null,
    comments: [],
    commentsError: null,
    commentsNumber: 0,
    addCommentStatus: null,
    deleteCommentStatus: null,
    status: null,
    sendRatingStatus: null,
    getRatingsStatus: null,
};

/* eslint-disable default-case, no-param-reassign */
const recipeContainerReducer = (state = initialState, action) =>
    produce(state, draft => {
        switch (action.type) {
            case GET_RATINGS:
                draft.getRatingsStatus = fetchStates.fetching;
                break;
            case GET_RATINGS_SUCCESS:
                draft.ratingsMean = action.ratingsMean;
                draft.message = action.message;
                draft.getRatingsStatus = fetchStates.success;
                break;
            case GET_RATINGS_ERROR:
                draft.error = action.error;
                draft.getRatingsStatus = fetchStates.error;
                break;
            case GET_RECIPE:
                // draft.status = fetchStates.fetching
                draft.recipeId = action.recipeId;
                break;
            case GET_RECIPE_SUCCESS:
                draft.recipe = action.recipe;
                draft.status = fetchStates.success;
                break;
            case GET_RECIPE_ERROR:
                draft.status = fetchStates.error;
                draft.error = action.error;
                break;
            // case GET_COMMENTS || GET_RATINGS:
            //   draft.status = fetchStates.fetching;
            //   break;
            case SEND_RATING_ERROR:
                draft.sendRatingStatus = fetchStates.error;
                break;
            case SEND_RATING:
                draft.sendRatingStatus = fetchStates.fetching;
                break;
            case SEND_RATING_SUCCESS:
                draft.sendRatingStatus = fetchStates.success;
                draft.ratingsMean = action.ratingsMean;
                break;
            case GET_COMMENTS_SUCCESS:
                // draft.status = fetchStates.success;
                draft.comments = action.response.comments;
                draft.commentsNumber = action.response.commentsNumber;
                break;
            case GET_COMMENTS_ERROR:
                // draft.status = fetchStates.error;
                draft.error = action.error;
                break;
            case ADD_COMMENT:
                draft.addCommentStatus = fetchStates.fetching;
                draft.commentsError = '';
                break;
            case ADD_COMMENT_SUCCESS:
                draft.addCommentStatus = fetchStates.success;
                break;
            case ADD_COMMENT_ERROR:
                draft.addCommentStatus = fetchStates.error;
                draft.commentsError = action.error;
                break;
            case REMOVE_COMMENT:
                draft.deleteCommentStatus = fetchStates.fetching;
                break;
            case REMOVE_COMMENT_SUCCESS:
                draft.deleteCommentStatus = fetchStates.success;
                break;
            case REMOVE_COMMENT_ERROR:
                draft.deleteCommentStatus = fetchStates.error;
                draft.commentsError = action.error;
                break;
            case CLEAR_STATUS:
                draft.addCommentStatus = null;
                break;
            case CLEAR_DELETE_COMMENTS_STATUS:
                draft.deleteCommentStatus = null;
                break;
            case CLEAR_SEND_RATING_STATUS:
                draft.sendRatingStatus = null;
                break;
        }
    });

export default recipeContainerReducer;
