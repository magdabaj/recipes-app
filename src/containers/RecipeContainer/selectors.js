import { createSelector } from 'reselect';
import { initialState } from './reducer';

/**
 * Direct selector to the recipeContainer state domain
 */

const selectRecipeContainerDomain = state => state.recipeContainer || initialState;

const selectRecipeIdRoute = (state, props) => props.match.params.recipeId;

/**
 * Other specific selectors
 */

/**
 * Default selector used by RecipeContainer
 */

const selectSlug = () =>
    createSelector(
        selectRecipeIdRoute,
        recipeId => recipeId,
    );

const makeSelectRecipeContainer = () =>
    createSelector(
        selectRecipeContainerDomain,
        substate => substate,
    );

const makeSelectRecipe = () =>
    createSelector(
        selectRecipeContainerDomain,
        substate => substate.recipe,
    );

const makeSelectRatingsMean = () =>
    createSelector(
        selectRecipeContainerDomain,
        substate => substate.ratingsMean,
    );

const makeSelectGetRatingsStatus = () =>
    createSelector(
        selectRecipeContainerDomain,
        substate => substate.getRatingsStatus,
    );

const makeSelectTags = () =>
    createSelector(
        selectRecipeContainerDomain,
        substate => substate.tags,
    );

const makeSelectComments = () =>
    createSelector(
        selectRecipeContainerDomain,
        substate => substate.comments,
    );

const makeSelectAddCommentStatus = () =>
    createSelector(
        selectRecipeContainerDomain,
        substate => substate.addCommentStatus,
    );

const makeSelectRemoveCommentStatus = () =>
    createSelector(
        selectRecipeContainerDomain,
        substate => substate.removeCommentStatus,
    );

const makeSelectAddCommentError = () =>
    createSelector(
        selectRecipeContainerDomain,
        substate => substate.commentsError,
    );

const makeSelectCommentsNumber = () =>
    createSelector(
        selectRecipeContainerDomain,
        substate => substate.commentsNumber,
    );

const makeSelectStatus = () =>
    createSelector(
        selectRecipeContainerDomain,
        substate => substate.status,
    );

export default makeSelectRecipeContainer;
export {
    selectRecipeContainerDomain,
    makeSelectRatingsMean,
    makeSelectRecipe,
    makeSelectGetRatingsStatus,
    makeSelectTags,
    selectSlug,
    makeSelectComments,
    makeSelectAddCommentStatus,
    makeSelectAddCommentError,
    makeSelectCommentsNumber,
    makeSelectRemoveCommentStatus,
    makeSelectStatus,
};
