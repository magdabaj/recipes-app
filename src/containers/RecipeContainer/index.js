/**
 *
 * RecipeContainer
 *
 */

import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { compose } from 'redux';

import makeSelectRecipeContainer, {
    makeSelectAddCommentError, makeSelectAddCommentStatus,
    makeSelectComments,
    makeSelectCommentsNumber, makeSelectDeleteCommentStatus,
    makeSelectGetRatingsStatus,
    makeSelectRatingsMean,
    makeSelectRecipe, makeSelectSendRatingStatus,
    makeSelectStatus,
    selectSlug,
} from './selectors';
import { makeSelectTags } from "../App/selectors";
import { makeSelectUser } from '../LoginContainer/selectors';
import {
    addComment,
    editComment,
    getComments,
    getRecipe,
    getRecipeRatings,
    removeComment,
    sendRating,
} from './actions';
import { loadTags } from "../App/actions";
import RecipeWrapper from '../../components/RecipeWrapper';
import fetchStates from '../../utils/fetchStates';
import Spinner from '../../components/Spinner';

export function RecipeContainer({ status, ...props }) {

    useEffect(() => {
        if (props.tags.length === 0) props.loadTags();
    }, []);

    useEffect(() => {
        if (props.recipeId && !props.ratingsMean) props.getRecipeRatings(props.recipeId);
    }, [
        /*props.recipeId*/
    ]);

    useEffect(() => {
        if (props.recipeId) props.getRecipe(props.recipeId);
    }, [props.recipeId]);

    useEffect(() => {
        props.getComments(props.recipeId);
    }, [props.addCommentStatus, props.deleteCommentStatus]);

    return status === fetchStates.fetching ? <Spinner /> : <RecipeWrapper {...props} />;
}

RecipeContainer.propTypes = {};

const mapStateToProps = createStructuredSelector({
    recipeContainer: makeSelectRecipeContainer(),
    user: makeSelectUser(),
    recipe: makeSelectRecipe(),
    ratingsMean: makeSelectRatingsMean(),
    getRatingsStatus: makeSelectGetRatingsStatus(),
    tags: makeSelectTags(),
    recipeId: selectSlug(),
    comments: makeSelectComments(),
    addCommentStatus: makeSelectAddCommentStatus(),
    deleteCommentStatus: makeSelectDeleteCommentStatus(),
    addCommentError: makeSelectAddCommentError(),
    commentsNumber: makeSelectCommentsNumber(),
    status: makeSelectStatus(),
    sendRatingStatus: makeSelectSendRatingStatus(),
});

function mapDispatchToProps(dispatch) {
    return {
        getRecipe: recipeId => dispatch(getRecipe(recipeId)),
        getRecipeRatings: recipeId => dispatch(getRecipeRatings(recipeId)),
        loadTags: () => dispatch(loadTags()),
        sendRating: (rate, userId, recipeId) => dispatch(sendRating(rate, userId, recipeId)),
        getComments: recipeId => dispatch(getComments(recipeId)),
        addComment: (recipeId, content, email) => dispatch(addComment(recipeId, content, email)),
        editComment: (commentId, userId, content) => dispatch(editComment(commentId, userId, content)),
        removeComment: commentId => dispatch(removeComment(commentId)),
    };
}

const withConnect = connect(
    mapStateToProps,
    mapDispatchToProps,
);

export default compose(withConnect)(RecipeContainer);
