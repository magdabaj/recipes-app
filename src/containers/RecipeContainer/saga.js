import { all, call, put, select, takeLatest } from 'redux-saga/effects';

import { createRating, fetchRecipeRatings } from '../../utils/api/rating';
import {
    getRecipeError,
    getRecipeRatingsError,
    getRecipeRatingsSuccess,
    getRecipeSuccess,
    sendRatingSuccess,
    sendRatingError,
    getCommentsError,
    getCommentsSuccess,
    addCommentError,
    addCommentSuccess,
    getComments,
    removeCommentSuccess,
    removeCommentError, clearStatus,
} from './actions';
import {
    ADD_COMMENT,
    GET_COMMENTS,
    GET_RATINGS,
    GET_RECIPE,
    REMOVE_COMMENT,
    SEND_RATING,
} from './constants';
import { fetchRecipe } from '../../utils/api/recipe';
import { deleteComment, fetchComments, storeComment } from '../../utils/api/comment';
import { makeSelectUser } from '../LoginContainer/selectors';

export function* handleRecipeRatingsLoad(action) {
    const { recipeId } = action;

    try {
        const { message, ratingsMean } = yield call(fetchRecipeRatings, recipeId);
        yield put(getRecipeRatingsSuccess(message, recipeId, ratingsMean));
    } catch (e) {
        yield put(getRecipeRatingsError(e.message));
    }
}

export function* watchRecipeRatingsLoad() {
    yield takeLatest(GET_RATINGS, handleRecipeRatingsLoad);
}

export function* handleRatingSend(action) {
    const { rate, userId, recipeId } = action;
    try {
        const json = yield call(createRating, rate, userId, recipeId);
        if (json.type === 'error') {
            yield put(sendRatingError(json.message));
        } else {
            yield put(sendRatingSuccess(json, recipeId, json.ratingsMean));
        }
    } catch (e) {
        yield put(sendRatingError(e.message));
    }
}

export function* watchRatingSend() {
    yield takeLatest(SEND_RATING, handleRatingSend);
}

function* handleRecipeLoad(action) {
    const { recipeId } = action;
    try {
        const recipe = yield call(fetchRecipe, recipeId);
        yield put(getRecipeSuccess(recipe));
    } catch (e) {
        yield put(getRecipeError(e.message));
    }
}

export function* watchRecipeLoad() {
    yield takeLatest(GET_RECIPE, handleRecipeLoad);
}

export function* loadCommentsSaga({ recipeId }) {
    try {
        const comments = yield call(fetchComments, recipeId);
        yield put(getCommentsSuccess(comments));
    } catch (e) {
        yield put(getCommentsError(e.message));
    }
}

export function* watchCommentsLoad() {
    yield takeLatest(GET_COMMENTS, loadCommentsSaga);
}

export function* addCommentSaga({ recipeId }) {
    const { userId } = yield select(makeSelectUser());
    try {
        const message = yield call(storeComment, recipeId.recipeId, recipeId.content, userId);
        yield put(addCommentSuccess(message));
        yield put(clearStatus());
    } catch (e) {
        yield put(addCommentError(e.message));
        yield put(clearStatus());
    }
}

export function* watchCommentAdd() {
    yield takeLatest(ADD_COMMENT, addCommentSaga);
}

export function* deleteCommentSaga(action) {
    const { commentId } = action;
    const { userId } = yield select(makeSelectUser());

    try {
        const message = yield call(deleteComment, commentId, userId);
        yield put(removeCommentSuccess(message));
        yield put(clearStatus());
    } catch (e) {
        yield put(removeCommentError(e.message));
        yield put(clearStatus());
    }
}

export function* watchCommentDelete() {
    yield takeLatest(REMOVE_COMMENT, deleteCommentSaga);
}

