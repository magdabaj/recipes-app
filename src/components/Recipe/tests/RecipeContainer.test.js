import React from "react";
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import RecipeContainer from "../RecipeContainer";

test('RecipeContainer', () => {
    const tree = renderer.create(<RecipeContainer/>).toJSON()
    expect(tree).toMatchSnapshot()
})
//
// test('RecipeContainer', () => {
//     const tree = render(<RecipeContainer/>)
//     // expect(tree).toMatchSnapshot()
//     console.log(tree)
//     // expect(tree.find('RecipeContainer')).toHaveStyleRule('display', 'flex')
// })