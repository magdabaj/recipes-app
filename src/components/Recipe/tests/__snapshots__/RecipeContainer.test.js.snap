// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeContainer 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin-bottom: 3vw;
  color: #5D001E;
}

@media (max-width:768px) {
  .c0 {
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

<div
  className="c0"
/>
`;
